src/v/iceberg/manifest_io.cc

```diff
@@ -53,30 +52,36 @@ manifest_io::download_manifest_list(const manifest_list_path& path) {
 ss::future<checked<size_t, metadata_io::errc>>
 manifest_io::upload_manifest(const manifest_path& path, const manifest& m) {
     return upload_object<manifest>(
-      path(), m, "iceberg::manifest", [](const manifest& m) {
+      path().string(), m, "iceberg::manifest", [](const manifest& m) {
           return serialize_avro(m);
       });
 }
 
+ss::future<checked<size_t, metadata_io::errc>>
+manifest_io::upload_manifest(const uri& uri, const manifest& m) {
+    return upload_manifest(manifest_path(from_uri(uri)), m);
+}
+
 ss::future<checked<size_t, metadata_io::errc>>
 manifest_io::upload_manifest_list(
   const manifest_list_path& path, const manifest_list& m) {
     return upload_object<manifest_list>(
-      path(), m, "iceberg::manifest_list", [](const manifest_list& m) {
+      path().string(), m, "iceberg::manifest_list", [](const manifest_list& m) {
           return serialize_avro(m);
       });
 }
 
-ss::sstring manifest_io::to_uri(const std::filesystem::path& p) const {
-    return fmt::format("{}{}", uri_base(), p.native());
+ss::future<checked<size_t, metadata_io::errc>>
+manifest_io::upload_manifest_list(const uri& uri, const manifest_list& m) {
+    return upload_manifest_list(manifest_list_path(from_uri(uri)), m);
+}
+
+uri manifest_io::to_uri(const std::filesystem::path& p) const {
+    return make_uri(bucket_, p);
 }
 
-std::filesystem::path manifest_io::from_uri(const ss::sstring& s) const {
-    const auto base = uri_base();
-    if (!s.starts_with(base)) {
-        return std::filesystem::path{s};
-    }
-    return std::filesystem::path{s.substr(base.size())};
+std::filesystem::path manifest_io::from_uri(const uri& s) const {
+    return std::filesystem::path(path_from_uri(s));
```

Comment:
Can we wrap this in a try/catch in case this throws, and have it return a result type?

Reply:
sure will do

Reply:
Thank you!